### PR TITLE
Web app start fix when config file is missing #1413

### PR DESCRIPTION
**Pull request by ozankiraz**
Created New PR since I messed up previous one

### CODE REVIEW

File: web/index.jsp [hunk 1]
```diff
@@ -29,6 +29,14 @@ include file="projects.jspf"
 {
     PageConfig cfg = PageConfig.get(request);
     cfg.setTitle("Search");
+    String sourceRootPath = cfg.getSourceRootPath();
+    if(sourceRootPath == null || sourceRootPath.isEmpty()){
+    throw new java.io.FileNotFoundException("Configuration File Not Found");
+    }
+    File sourceRootPathFile = RuntimeEnvironment.getInstance().getSourceRootFile();
+    if(sourceRootPathFile.exists() && !sourceRootPathFile.canRead()){
```
Comment (tulinkry) [inline web/index.jsp]: This is not correct
```java
if(sourceRootPathFile.exists() && !sourceRootPathFile.canRead()){
```

Perhaps should have been:
```java
if(!sourceRootPathFile.exists() || !sourceRootPathFile.canRead()){
```

Comment (ozankiraz) [inline web/index.jsp]: @tulinkry I intended to do this actually, There are two exceptions, first one is FileNotFoundException, thrown when source path is null or empty, covers the scenario when source root path is not set,
Second exception is IOException, covers the scenario when file is there but cannot read

Comment (tulinkry) [inline web/index.jsp]: But when the file does not actually exist then there is no exception which is not correct, is it?

File: src/org/opensolaris/opengrok/web/PageConfig.java [hunk 2]
```diff
@@ -1402,4 +1402,20 @@ public String getPathTitle() {
 
         return Util.htmlize(title);
     }
+    
+    public void checkSourceRootExistence() throws IOException {
```
Comment (ozankiraz) [inline src/org/opensolaris/opengrok/web/PageConfig.java]: Moved logic here

File: src/org/opensolaris/opengrok/web/PageConfig.java [hunk 3]
```diff
@@ -1402,4 +1402,20 @@ public String getPathTitle() {
...
+        if (sourceRootPath == null || sourceRootPath.isEmpty()) {
+            throw new java.io.FileNotFoundException("Configuration File Not Found, Can not determine source root path");
+        }
+        
+        File sourceRootPathFile = RuntimeEnvironment.getInstance().getSourceRootFile();
+        if (sourceRootPathFile.exists() && !sourceRootPathFile.canRead()) {
+            throw new java.io.IOException("Can not read configuration file");
+        }
+        
+        if(!sourceRootPathFile.exists()){
```
Comment (ozankiraz) [inline src/org/opensolaris/opengrok/web/PageConfig.java]: Added this check as required

File: web/index.jsp [hunk 4]
```diff
@@ -29,14 +29,7 @@ include file="projects.jspf"
...
     cfg.setTitle("Search");
-    String sourceRootPath = cfg.getSourceRootPath();
-    if(sourceRootPath == null || sourceRootPath.isEmpty()){
-    throw new java.io.FileNotFoundException("Configuration File Not Found");
-    }
-    File sourceRootPathFile = RuntimeEnvironment.getInstance().getSourceRootFile();
-    if(sourceRootPathFile.exists() && !sourceRootPathFile.canRead()){
-    throw new java.io.IOException("Can not read configuration file");
-    }
+    cfg.checkSourceRootExistence();
```
Comment (ozankiraz) [inline web/index.jsp]: Shorter and more readable

**Comment by ozankiraz** [comment]
@tulinkry Yes, you're right, I was confused about SRC_ROOT in web.xml and sourceRoot in configuration.xml
Moved logic to PageConfig class and updated jsp file. Thanks for pointing it out.

**Comment by tulinkry** [comment]
Ok to me now

**Comment by vladak** [comment]
@ozankiraz any news w.r.t. your OCA ?

**Comment by ozankiraz** [comment]
Hi @vladak 
I got an email about OCA is processed and I can see my name in OCA list, but in email they didn't sent me OCA number.

**Comment by vladak** [comment]
Okay, I see your name there, merging.

**Pull request merged**